src/app_service/service/wallet_account/service.nim

```diff
@@ -274,6 +274,35 @@ QtObject:
       return
     result = self.walletAccounts[address]
 
+  proc getAccountsByAddresses*(self: Service, addresses: seq[string]): seq[WalletAccountDto] = 
+    for address in addresses:
+      result.add(self.getAccountByAddress(address))
+
+  proc getTokensByAddresses*(self: Service, addresses: seq[string]): seq[WalletTokenDto] =
+    var tokens = initTable[string, WalletTokenDto]()
+    for address in addresses:
+      let walletAccount = self.getAccountByAddress(address)
+      for token in walletAccount.tokens:
+        if not tokens.hasKey(token.symbol):
+          let newToken = token.copyToken()
+          tokens[token.symbol] = newToken
+          continue
+
+        for chainId, balanceDto in token.balancesPerChain:
+          if not tokens[token.symbol].balancesPerChain.hasKey(chainId):
```

Comment:
I think in this specific case this might be more readable. (Continue used above is fine).
It's just personal, feel free to ignore it if all code uses same convention (I'm still trying to get grasp on that :) )
```suggestion
          if tokens[token.symbol].balancesPerChain.hasKey(chainId):
            tokens[token.symbol].balancesPerChain[chainId].balance += balanceDto.balance
          else:
            tokens[token.symbol].balancesPerChain[chainId] = BalanceDto(
              balance: balanceDto.balance,
              address: balanceDto.address,
              chainId: balanceDto.chainId,
              hasError: balanceDto.hasError,
            )
```